opendevin/controller/agent_controller.py

```diff
@@ -225,6 +225,11 @@ async def set_agent_state_to(self, new_state: AgentState):
         ):
             # user intends to interrupt traffic control and let the task resume temporarily
             self.state.traffic_control_state = TrafficControlState.PAUSED
+            # User has chosen to deliberately continue - lets double the max iterations
+            if self.state.iteration >= self.state.max_iterations:
+                self.state.max_iterations *= 2
+            if self.state.metrics.accumulated_cost >= self.max_budget_per_task:
+                self.max_budget_per_task *= 2
```

Comment:
This piece of code alone doesn't work in a multi-agent context. This essentially expands the current agent/controller's limit, but when it returns to its parent, the old limit would be hit immediately. It would still work but a little bit confusing/surprising to users. The backend logs would look weird, too.

I would suggest take a look at code in `if delegate_done` section. You probably need to do something similar to "propagate back" the `max_iterations` and `max_budget_per_task`.